src/mappings/poktroll/suppliers.ts

```diff
@@ -0,0 +1,257 @@
+  logger.debug(`[handleSupplierStakeMsg] (msg.msg): ${stringify(msg.msg, undefined, 2)}`);
+
+  if (!msg.msg.decodedMsg.stake) {
+    return logger.error(`[handleSupplierStakeMsg] stake not provided in msg`);
+  }
+
+  const msgId = messageId(msg)
+
+  const supplierMsgStake = MsgStakeSupplierEntity.create({
+    id: msgId,
+    signerId: msg.msg.decodedMsg.signer,
+    supplierId: msg.msg.decodedMsg.operatorAddress,
+    ownerId: msg.msg.decodedMsg.ownerAddress,
+    stakeAmount: BigInt(msg.msg.decodedMsg.stake.amount),
+    stakeDenom: msg.msg.decodedMsg.stake.denom,
+    blockId: msg.block.block.id,
+    transactionId: msg.tx.hash,
+    messageId: msgId,
+  })
+
+  const supplier = Supplier.create({
+    id: msg.msg.decodedMsg.operatorAddress,
+    operatorId: msg.msg.decodedMsg.operatorAddress,
+    ownerId: msg.msg.decodedMsg.ownerAddress,
+    stakeAmount: BigInt(msg.msg.decodedMsg.stake.amount),
+    stakeDenom: msg.msg.decodedMsg.stake.denom,
+    status: StakeStatus.Staked,
+    unstakingEndHeight: undefined,
+    unstakingEndBlockId: undefined,
+    unstakingBeginBlockId: undefined,
+  })
+
+  const servicesId: Array<string> = []
+  const supplierMsgStakeServices: Array<MsgStakeSupplierServiceProps> = []
+  const newSupplierServices: Array<SupplierServiceConfigProps> = []
+
+  const operatorAddress = msg.msg.decodedMsg.operatorAddress
+
+  for (const {endpoints, revShare, serviceId} of msg.msg.decodedMsg.services) {
+    servicesId.push(serviceId)
+
+    const endpointsArr = endpoints.map((endpoint) => ({
+      url: endpoint.url,
+      rpcType: endpoint.rpcType,
+      configs: endpoint.configs,
+    }))
+
+    const revShareArr = revShare.map((revShare) => ({
+      address: revShare.address,
+      revSharePercentage: revShare.revSharePercentage,
+    }))
+
+    supplierMsgStakeServices.push({
+      id: getMsgStakeServiceId(msgId, serviceId),
+      serviceId,
+      stakeMsgId: msgId,
+      endpoints: endpointsArr,
+      revShare: revShareArr
+    })
+
+    newSupplierServices.push({
+      id: getStakeServiceId(operatorAddress, serviceId),
+      serviceId,
+      supplierId: operatorAddress,
+      endpoints: endpointsArr,
+      revShare: revShareArr
+    })
+  }
+
+  const currentSupplierServices = await SupplierServiceConfig.getBySupplierId(operatorAddress, {})
+
+  const servicesToRemove: Array<string> = []
+
+  if (currentSupplierServices && currentSupplierServices.length > 0) {
+    for (const service of currentSupplierServices) {
+      if (!servicesId.includes(service.serviceId)) {
+        servicesToRemove.push(service.id)
+      }
+    }
+  }
+
+  const promises: Array<Promise<void>> = [
+    supplierMsgStake.save(),
+    supplier.save(),
+    store.bulkCreate('MsgStakeSupplierService', supplierMsgStakeServices),
+    store.bulkCreate('SupplierServiceConfig', newSupplierServices),
+  ]
+
+  if (servicesToRemove.length > 0) {
+    promises.push(store.bulkRemove('SupplierServiceConfig', servicesToRemove))
+  }
```

Comment:
1. We can't have arrays in the entities.
2. This remove is a soft one, it just close the block range so it will still exists in the historical state.